---
title: "R Notebook"
output: html_notebook
---
```{r}

library(DESeq2)
library(ggplot2)
library(gplots)
library(tidyverse)
library(dplyr)
library(viridisLite)

```

```{r}

# Read in your counts file that has already been dictated to add the gene
# symbol onto the ENSEMBL ID.

Counts <- read.csv('named.samples.parsed.csv', header = TRUE)
Counts <- Counts %>% distinct(Gene, .keep_all = TRUE)
row.names(Counts) <- Counts$Gene
Counts[1] <- NULL
Counts

```

```{r}
# Eliminate any reads without counts

Counts <- Counts[which(rowSums(Counts) > 0) , ]
Counts


```

```{r}
# Create a vector that contains the conditions used in the experiment

condition <- factor(c("Con","Con","Con","Con","Con","Con", "Pio","Pio","Pio","Pio","Pio","Pio"))
coldata <- data.frame(row.names = colnames(Counts), condition)
coldata

```

```{r}

# Create a DESeq data set and tell it how to evaluate the conditions

dds <- DESeqDataSetFromMatrix(countData = Counts, colData = coldata, design = ~condition)


```

```{r}
# Perform DESeq on the dds DESeq matrix
dds <- DESeq(dds)
dds

```

```{r}
# Use the vst function to evaulate the sequencing

vsdata <- vst(dds, blind = FALSE)

```

```{r}
# Evaluate the PCA of the experiment

plotPCA(vsdata, intgroup = "condition")

```

```{r}

# Plot dispersion analysis

plotDispEsts(dds)

```

```{r}

# Run differential analysis on the DESeq object

res <- results(dds, contrast = c("condition", "Con", "Pio"))
res

```

```{r}

# Cull the list by getting rid of NA readings and keeping only signficant genes

sigs <- na.omit(res)
sigs <- sigs[sigs$padj < 0.05, ]
sigs

```

```{r}
# Return only highly represented genes and genes with a fold change greater than 2.

topsigs <- sigs [sigs$baseMean > 300 & (abs(sigs$log2FoldChange) > 2), ]
topsigs


```

```{r}

# Order this list 

topsigs<- topsigs[order(topsigs$log2FoldChange, decreasing = TRUE), ]
topsigs

```

```{r}
# Extract the rLog values for the DESeq set

rlog_out <- rlog(dds, blind = FALSE)
```



```{r}
# Assay this vector using DESeq2

mat<-assay(rlog_out)[rownames(topsigs), rownames(coldata)]
mat
```


```{r}
# Get rid of any NA readings
mat <- na.omit(mat)

```


```{r}
# Scale the data matrix
mat.scaled <- t(apply(mat, 1, scale))
mat.scaled
```


```{r}
# Assign the conditions to the matrix
colnames(mat) <- coldata$condition
colnames(mat.scaled)<-colnames(mat)
mat.scaled

```
```{r}
#Grab the top and bottom genes and bind them in one matrix
mat.top <- mat.scaled[1:25, 1:12]
mat.bottom <- mat.scaled[353:378, 1:12]
matbind <- rbind(mat.top,mat.bottom)
matbind

```
```{r}
heatmap.2( matbind, 
           scale="row", 
           trace="none", 
           dendrogram="both",
           density.info="none",
           key.title= "Z-score",
           key.ylab="none",
           key.xlab="",
       
           main ="APCmin Organoids Pioglitazone RNASeq",
     
           col = heat.colors(100),
           ColSideColors = c( Control="gray", DPN="blue", OHT="orange" )[
           colData(rlog_out)$condition ] )
```

